### ISSUE TITLE
Enhancement get top left and bottom right postition of visible region

### ISSUE THREAD

**Issue by deextercz** [amay077/Xamarin.Forms.GoogleMaps#94]
I want to know if my custom pin is in range or not.

**Comment by amay077** [comment]
Is ``MapSpan`` not good enough?

``MapSpan`` has Center, LatitudeDegrees(as height) and LongitudeDegrees(as width).
You can calculate top/left/bottom/right straightforward.

**Comment by deextercz** [comment]
Yes I agree, I just wanted to call method on map like this:
class Map
public bool IsInVisibleRegion(MapSpan point)

So in general my function should provide this calculation.
I have code changes already prepared, but I am not able to push pullrequest

**Comment by amay077** [comment]
Sorry I quite forget important thing. 

I should consider map tilting and rotating.
With these taken into consideration, ``IsInVisibleRegion``'s implementation will be difficult.

On the other hand, Native API has [VisibleRegion](https://developers.google.com/android/reference/com/google/android/gms/maps/model/VisibleRegion) and [LatLngBounds](https://developers.google.com/android/reference/com/google/android/gms/maps/model/LatLngBounds).

VisibleRegion has 4 points(nearLeft, nearRight, farLeft and farRight) and can get "smallest bounding" as LatLngBounds.
LatLngBounds has [``contains``](https://developers.google.com/android/reference/com/google/android/gms/maps/model/LatLngBounds.html#contains(com.google.android.gms.maps.model.LatLng)) method.

In conclusion, I will plan to wrapping VisibleRegion class and add ``Contains(latlng)`` to [``Bounds``](https://github.com/amay077/Xamarin.Forms.GoogleMaps/blob/master/Xamarin.Forms.GoogleMaps/Xamarin.Forms.GoogleMaps/Bounds.cs) class in #7. 

If your case is not applicable, you could use your own implementation.

**Comment by deextercz** [comment]
Thank you this will be enough for me.

**Comment by amay077** [comment]
This discussion has been moved to #7 .
